ambry-router/src/main/java/com/github/ambry/router/GetBlobInfoOperation.java

```diff
@@ -356,6 +360,7 @@ private void handleBody(InputStream payload, MessageMetadata messageMetadata, Me
     if (encryptionKey == null) {
       // if blob is not encrypted, move the state to Complete
       BlobInfo blobInfo = new BlobInfo(serverBlobProperties, userMetadata.array(), messageInfo.getLifeVersion());
+      handleQuotaEvent();
```

Comment:
nit: can we do this in `checkAndMaybeComplete` method, just like what we did in `DeleteOperation`.

Reply:
+1 let's centralize on checkAndMaybeComplete with extra state checking logic to ensure the callback is called only once.